src/StimulusBundle/src/Dto/StimulusAttributes.php

```diff
@@ -108,18 +108,18 @@ public function addAttribute(string $name, string $value): void
     public function __toString(): string
     {
         $controllers = array_map(function (string $controllerName): string {
-            return $this->escapeAsHtmlAttr($controllerName);
+            return $this->escape($controllerName, 'html');
         }, $this->controllers);
 
         // done separately so we can escape, but avoid escaping ->
         $actions = array_map(function (array $actionData): string {
-            $controllerName = $this->escapeAsHtmlAttr($actionData['controllerName']);
-            $actionName = $this->escapeAsHtmlAttr($actionData['actionName']);
+            $controllerName = $actionData['controllerName'];
+            $actionName = $actionData['actionName'];
             $eventName = $actionData['eventName'];
 
-            $action = $controllerName.'#'.$actionName;
+            $action = $this->escape($controllerName.'#'.$actionName, 'html');
```

Comment:
It keeps the same behaviour, but with **one** `$this->escape()` call, `#` is still rendered as `#`.